river/feature_extraction/agg.py

```diff
@@ -126,26 +120,33 @@ class Agg(base.Transformer):
     """
 
     def __init__(
-        self, on: str, by: typing.Union[str, typing.List[str]], how: stats.Univariate
+        self,
+        on: str,
+        by: typing.Optional[typing.Union[str, typing.List[str]]],
```

Comment:
`by` should  be set to `None` by default when using `typing.Optional` in annotation ?

Reply:
I thought about that too. Not sure. I don't think that `typing.Optional` forces us to put a `None` by default. It just hints that we can pass a `None`. I rather have the user explicitly pass the `None` themselves, because not using a group by key is not a common behavior. Let me know if you disagree!

Reply:
I understand your point of view and I agree with you.
Using `aggregate` without `by` is "unusual" behavior.